Introduce Undefined::VALUE

oprypkhantc opened the pull request:
Closes #726 

Tests are incomplete, but otherwise this is ready for review

From my perspective, the next step would be deprecating `isUpdate` completely - what do you think of that? I didn't include it in this PR as there may be value in trying out the `Undefined` approach first for a larger chunk of users. For us though, this approach has been good.

oprypkhantc committed:
Introduce Undefined::VALUE

oprypkhantc committed:
Fix failing tests

oojacoboo commented:
Thanks for putting this together @oprypkhantc!

I'm thinking we should deprecate support for `InputType::isUpdate` as part of this PR and update the docs to direct the `Undefined` "type" as the de facto way of handling update functionality, where nullable support is needed.

oojacoboo commented:
@oprypkhantc can we get this merged?

oprypkhantc committed:
Merge branch 'master' of github.com:thecodingmachine/graphqlite into undefined

# Conflicts:
#	src/Mappers/Parameters/TypeHandler.php

oprypkhantc committed:
Deprecate update flag, update the docs

oprypkhantc commented:
Hey @oojacoboo. Sorry, I forgot to update you - I no longer work directly for the company I used to and no longer work with PHP; I might occasionally come back and fix things here and there though :)

I updated the PR with deprecation of `update: ` flag and updated the docs, but feel free to take over if I missed anything; also if you need me for anything you can tag my personal account at @autaut03, I'll see a notification for that one ❤️

oojacoboo committed:
Add trigger_error function import to Input.php

oojacoboo commented:
@oprypkhantc @autaut03 Thanks for the update, both on the PR and your personal situation.  It's been a pleasure working with you on graphqlite.  All the best in your new endeavors!
